### ISSUE TITLE
Cannot invoke "org.eclipse.jetty.io.Content$Chunk.hasRemaining()" because "chunk" is null

### ISSUE THREAD

**Issue by olenagerasimova** [jetty-project/jetty-reactive-httpclient#296]
This test code with version **4.0.0** of jetty-reactive-httpclient on java 20
```
import io.reactivex.Flowable;
import java.nio.ByteBuffer;
import java.util.concurrent.CountDownLatch;
import java.util.concurrent.TimeUnit;
import org.eclipse.jetty.client.HttpClient;
import org.eclipse.jetty.reactive.client.ReactiveRequest;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class MavenTest {

    @Test
    void test() throws Exception {
        final HttpClient client = new HttpClient();
        client.start();
        final CountDownLatch contentLatch = new CountDownLatch(1);
        final CountDownLatch responseLatch = new CountDownLatch(1);
        final ReactiveRequest rq = ReactiveRequest.newBuilder(
            client.newRequest("https://repo.maven.apache.org/maven2/args4j/args4j/2.32/args4j-2.32.jar")
        ).build();
        Flowable.fromPublisher(
            rq.response(
                (rs, chunks) -> Flowable.fromPublisher(chunks).map(
                    chunk -> {
                        ByteBuffer byteBuffer = chunk.getByteBuffer();
                        chunk.release();
                        return byteBuffer.remaining();
                    }
                ).doOnComplete(contentLatch::countDown)
            )
        ).doOnComplete(responseLatch::countDown).subscribe();
        assertTrue(responseLatch.await(5, TimeUnit.SECONDS));
    }
}
```
leads to the following exception
```
[HttpClient@1623b78d-37] WARN org.eclipse.jetty.util.thread.SerializedInvoker - Serialized invocation error
java.lang.NullPointerException: Cannot invoke "org.eclipse.jetty.io.Content$Chunk.hasRemaining()" because "chunk" is null
	at org.eclipse.jetty.client.transport.ResponseListeners$ContentSourceDemultiplexer$ContentSource.onChunk(ResponseListeners.java:605)
	at org.eclipse.jetty.client.transport.ResponseListeners$ContentSourceDemultiplexer.onDemandCallback(ResponseListeners.java:492)
	at org.eclipse.jetty.util.thread.SerializedInvoker$Link.run(SerializedInvoker.java:191)
	at org.eclipse.jetty.util.thread.SerializedInvoker.run(SerializedInvoker.java:117)
	at org.eclipse.jetty.client.transport.HttpReceiver$ContentSource.invokeDemandCallback(HttpReceiver.java:722)
	at org.eclipse.jetty.client.transport.HttpReceiver$ContentSource.onDataAvailable(HttpReceiver.java:661)
	at org.eclipse.jetty.client.transport.HttpReceiver.responseContentAvailable(HttpReceiver.java:303)
	at org.eclipse.jetty.client.transport.internal.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:82)
	at org.eclipse.jetty.client.transport.internal.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:90)
	at org.eclipse.jetty.client.transport.internal.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:194)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:322)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
	at org.eclipse.jetty.io.ssl.SslConnection$SslEndPoint.onFillable(SslConnection.java:554)
	at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:373)
	at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:146)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:478)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:441)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:293)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:201)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:410)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:969)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1194)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1149)
	at java.base/java.lang.Thread.run(Thread.java:1623)
```

Please, take a look.

**Comment by sbordet** [comment]
Thank you for the report, it's a Jetty issue: https://github.com/jetty/jetty.project/issues/10726.

**Comment by sbordet** [comment]
Fixed with the update to Jetty 12.0.3.
